app/api/dao/user.py

```diff
@@ -170,14 +172,19 @@ def list_users(
 
         """
 
-        users_list = (
-            UserModel.query.filter(
-                UserModel.id != user_id,
-                not is_verified or UserModel.is_email_verified,
-                func.lower(UserModel.name).contains(search_query.lower())
-                | func.lower(UserModel.username).contains(search_query.lower()),
+        users_list_query = UserModel.query.filter(
+            UserModel.id != user_id,
+            not is_verified or UserModel.is_email_verified,
+            func.lower(UserModel.name).contains(search_query.lower())
+            | func.lower(UserModel.username).contains(search_query.lower()),
+        )
+        if skill != "":
+            users_list_query = users_list_query.filter(
+                func.lower(UserModel.skills) == func.lower(skill)
```

Comment:
Like with a wildcard at the beginning will not use indexes. 
Also porter will match Frustrate to Frustration which would be nice to have since skills isn't an enum.